### PR TITLE
Remove pwsh

### PR DESCRIPTION
**Pull request by JimMadge**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [x] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).

### :arrow_heading_up: Summary

Remove pwsh stuff, preparing to merge v5.0.0 work into develop branch

### CODE REVIEW

**Review by jemrobinson — CHANGES_REQUESTED** [review]
Can we hold off on this until 4.2.0 is finalised? I'd like to merge `main` into `python-migration` once that's done (so we get the updated docs etc.) and merging this PR will cause a lot of conflicts when we do that.

**Comment by JimMadge** [comment]
Hmm, I think we will have conflicts either way. Unless we close this and repeat the steps again later.

I don't think I'm very worried about conflicts here, they should mostly be "change vs remove" which is pretty straightforward. (I'm volunteering to manage the conflicts :smile:)

**Comment by JimMadge** [comment]
@craddm what do you think?

**Comment by craddm** [comment]
Is leaving it in place until v4.2.0 going to cause any problems? Inclined to leave as is, so that develop can be brought back to parity with v4.2.0 beforehand. Also aren't there a couple of potential issues we may want to do as a 4.2.1 that still need the powershell codebase?

**Comment by jemrobinson** [comment]
My view is that we should do things in this order:

1. get 4.2.0 tagged from `develop`
1. merge `develop` into `latest`
1. merge `develop` into `python-migration`
1. remove unnecessary things from `python-migration` and/or port any fixes that have been made in Powershell <- THIS PR
1. merge `python-migration` into `develop`

**Comment by JimMadge** [comment]
> Is leaving it in place until v4.2.0 going to cause any problems?

Nope.

> so that develop can be brought back to parity with v4.2.0 beforehand.

I think you are both keen on this. I can't really see the benefit, especially as we are going to immediately merge `python-migration` into `develop` anyway. Not a hill I will die on though so we can leave this PR open.

> Also aren't there a couple of potential issues we may want to do as a 4.2.1 that still need the powershell codebase?

Yes but those changes _won't_ be made on `develop`. That branch can start from v4.2.0

> 1. get 4.2.0 tagged from develop
> 1. merge develop into latest
> 1. merge develop into python-migration

I think there are a few things wrong with that (hill I will die on)
The process should be

1. Create release branch from `develop`
1. Apply any changes needed for release
1. Merge release branch into `latest`
1. Push tag to `latest`
1. Merge latest into `develop`

_I.e._ [gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)

Otherwise,

- The tag is on your `develop` branch, not the head of `latest`
- You merge changes from `develop` not on your release branch into `latest` (which in our case means changes which haven't been tested!)

**Comment by jemrobinson** [comment]
I agree with the release-ordering things that @JimMadge said :)

**Comment by jemrobinson** [comment]
We need to confirm that #1771 is done before merging this.

**Comment by JimMadge** [comment]
> We need to confirm that #1771 is done before merging this.

:+1: After that is merged this can be rebased/merged and conflicts addressed.

File: .devcontainer/python/Dockerfile
Comment (craddm) [inline .devcontainer/python/Dockerfile:1]: I'd rather you didn't remove these python devcontainer files, as they're still useful for the pulumi release - I use them when working on the python/pulumi version. They could be moved to the top-level .devcontainer directory, though.

They'll need some updating to reflect whatever version of Pulumi is being used, but I can sort that out when I come to work on it again.

File: .devcontainer/python/devcontainer.json
Comment (craddm) [inline .devcontainer/python/devcontainer.json:1]: See above comment

**Review by craddm — CHANGES_REQUESTED** [review]
LGTM, would keep the devcontainer though.

**Comment by JimMadge** [comment]
> LGTM, would keep the devcontainer though.

:100: Didn't see the devcontainer files for Python (or, they got deleted in a merge and I wasn't paying attention :smile:)

**Review by craddm — APPROVED** [review]

**Comment by jemrobinson** [comment]
Um, I think we missed the part where we needed to do #1771 (this is an issue not a PR) before merging this. Can we work on this next so it doesn't get missed?

**Comment by JimMadge** [comment]
Oops, I assumed that was the dev -> python_migration PR 🤦 . Sorry, yes let's prioritise that. The updated configuration is all still in the commit history.

**Pull request merged**
